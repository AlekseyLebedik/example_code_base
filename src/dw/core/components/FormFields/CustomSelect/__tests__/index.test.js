import React from 'react';
import { mount, shallow } from 'enzyme';

import Component from '../index';

describe('CustomSelect', () => {
  it('renders default values', () => {
    const props = {
      input: {},
      meta: {
        touched: undefined,
        error: undefined,
      },
      options: [],
    };
    expect(shallow(<Component {...props} />)).toMatchSnapshot();
  });

  it('renders custom values', () => {
    const props = {
      input: {},
      meta: {
        touched: true,
        error: 'invalid',
      },
      fullWidth: true,
      label: 'Blah blah',
      name: 'blahBlah',
      options: [
        { value: '1', label: 'label1' },
        { value: '2', label: 'label2' },
      ],
    };
    expect(shallow(<Component {...props} />)).toMatchSnapshot();
  });

  it('test events', () => {
    const props = {
      input: {},
      meta: {
        touched: undefined,
        error: undefined,
      },
      options: [],
    };
    const wrapper = mount(<Component {...props} />);
    const selectProps = wrapper.find('Select').first().props();
    selectProps.onFocus();
    expect(wrapper.state('focus')).toBe(true);

    selectProps.onBlur();
    expect(wrapper.state('focus')).toBe(false);
  });
});
