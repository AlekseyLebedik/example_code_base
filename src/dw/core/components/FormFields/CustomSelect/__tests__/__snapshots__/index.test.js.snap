// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect renders custom values 1`] = `
<div
  className="common-multi-select-field with-error"
>
  <Redux-form-ANTDundefined
    data-cy="titleEnvironmentSelector"
    fullWidth={true}
    getPopupContainer={[Function]}
    hasFeedback={false}
    input={
      Object {
        "onBlur": [Function],
        "onFocus": [Function],
      }
    }
    label="Blah blah"
    meta={
      Object {
        "error": "invalid",
        "touched": true,
      }
    }
    name="blahBlah"
    optionFilterProp="children"
    options={
      Array [
        Object {
          "disabled": false,
          "label": "label1",
          "value": "1",
        },
        Object {
          "disabled": false,
          "label": "label2",
          "value": "2",
        },
      ]
    }
    placeholder="Please select"
  />
  <div
    className="dropdown-placeholder ant-form-explain empty"
  >
    EMPTY
  </div>
</div>
`;

exports[`CustomSelect renders default values 1`] = `
<div
  className="common-multi-select-field"
>
  <Redux-form-ANTDundefined
    data-cy="titleEnvironmentSelector"
    getPopupContainer={[Function]}
    hasFeedback={false}
    input={
      Object {
        "onBlur": [Function],
        "onFocus": [Function],
      }
    }
    meta={
      Object {
        "error": undefined,
        "touched": undefined,
      }
    }
    optionFilterProp="children"
    options={Array []}
    placeholder="Please select"
  />
  <div
    className="dropdown-placeholder ant-form-explain empty"
  >
    EMPTY
  </div>
</div>
`;
